cache/refresher.go

```diff
@@ -238,3 +245,45 @@ func (s Refresher) handleDeleteFeatureEvent(ctx context.Context, env, identifier
 	// remove deleted flag entry.
 	return s.flagRepo.Remove(ctx, env, identifier)
 }
+
+func (s Refresher) handleFetchSegmentEvent(ctx context.Context, env, id string) error {
+	s.log.Debug("updating featureConfig entry", "environment", env, "identifier", id)
+
+	segmentConfig, err := s.clientService.FetchSegmentConfigForEnvironment(ctx, s.config.Token(), env)
+	if err != nil {
+		return err
+	}
+	segments := make([]domain.Segment, 0, len(segmentConfig))
+	for _, v := range segmentConfig {
+		segments = append(segments, domain.Segment(v))
+	}
+
+	// set the config
+	return s.segmentRepo.Add(ctx, domain.SegmentConfig{
+		EnvironmentID: env,
+		Segments:      segments,
+	})
+}
+
+func (s Refresher) handleDeleteSegmentEvent(ctx context.Context, env, identifier string) error {
```

Comment:
See if we get a Delete event, do we need to do a fetch from SaaS again? Was just thinking if we have the env and segment identifier can we not just remove it from the cache?

Reply:
Updated both Feature and Segment functions to update the config environment entry without making additional api call.